Source/Basic Shapes/SvgVisualElement.cs

```diff
@@ -279,41 +279,87 @@ protected internal virtual bool RenderStroke(ISvgRenderer renderer)
                         {
                             using (var pen = new Pen(brush, strokeWidth))
                             {
-                                if (this.StrokeDashArray != null && this.StrokeDashArray.Count > 0)
+                                if (StrokeDashArray != null && StrokeDashArray.Count > 0)
                                 {
-                                    if (this.StrokeDashArray.Count % 2 != 0)
+                                    strokeWidth = strokeWidth <= 0 ? 1 : strokeWidth;
+                                    if (StrokeDashArray.Count % 2 != 0)
                                     {
                                         // handle odd dash arrays by repeating them once
-                                        this.StrokeDashArray.AddRange(this.StrokeDashArray);
+                                        StrokeDashArray.AddRange(StrokeDashArray);
                                     }
 
-                                    /* divide by stroke width - GDI behaviour that I don't quite understand yet.*/
-                                    pen.DashPattern = this.StrokeDashArray.Select(u => ((u.ToDeviceValue(renderer, UnitRenderingType.Other, this) <= 0) ? 1 : u.ToDeviceValue(renderer, UnitRenderingType.Other, this)) /
-                                        ((strokeWidth <= 0) ? 1 : strokeWidth)).ToArray();
+                                    SvgUnit dashOffset = StrokeDashOffset != null ? StrokeDashOffset : 0;

```

Comment:
Use `var` ?

Reply:
Doesn't work with 0, I think I tried that.

Reply:
Use `SvgUnit.Empty` !

Reply:
Well, turns out it works with 0 too - I was wrong!

Reply:
Done.